easybuild/easyconfigs/r/ReFrame/ReFrame-4.6.2.eb

```
@@ -0,0 +1,87 @@
+easyblock = 'PythonBundle'
+
+name = 'ReFrame'
+version = '4.6.2'
+
+homepage = 'https://github.com/reframe-hpc/reframe'
+description = '''ReFrame is a framework for writing regression tests for HPC systems.'''
+
+toolchain = SYSTEM
+
+allow_system_deps = [('Python', SYS_PYTHON_VERSION)]
+
+# Required by lxml, which is installed by ReFrame's bootstrap installer
+osdependencies = [
+    ('libxml2'),
+    ('libxslt'),
```

```suggestion
    ('libxslt', 'libxslt1.1'),
```

This solves the issue with libxslt, but now the installation of setuptools fails:
```
  File "/data/eb/build/ReFrame/4.6.2/system-system/setuptools/setuptools-59.6.0/pkg_resources/__init__.py", line 2172, in <module>
    register_finder(pkgutil.ImpImporter, find_on_path)
                    ^^^^^^^^^^^^^^^^^^^
AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
```
This has been removed in Python 3.12 (see https://docs.python.org/3/whatsnew/3.12.html#importlib), which is the default Python on Ubuntu 24.04. But bumping the setuptools version would break Python 3.6 support...

---

This, exactly this stuff is the problem with a ReFrame at `SYSTEM` toolchain level. But I know some people still really prefer it, which is why I tried to push one (but with proper osdeps and everything).

Ok, so we have two choices: support 3.6-3.11, or bump setuptools and support 3.7-3.12? How far would you need to bump setuptools, would it still support 3.7? I see 68.0.0 is the latest that supports 3.7, is that high enough to resolve your issue?

---

Seems to have been fixed in `66.1.0` https://github.com/pypa/setuptools/commit/6653e747c3815b140156249205397ef3719581ee . That would give us a small window of versions that should work with both 3.7 and 3.12...